File: 0001-0500/0038/0038.count-and-say.cpp
/*
 * @lc app=leetcode id=38 lang=cpp
 *
 * [38] Count and Say
 */

// @lc code=start
class Solution
{
public:
    string countAndSay(int n)
    {
        if (n == 1)
            return "1";
        string s = countAndSay(n - 1);
        string res = "";
        int count = 1;
        for (int i = 0; i < s.size(); i++)
        {
            if (i + 1 < s.size() && s[i] == s[i + 1])
            {
                count++;
            }
            else
            {
                res += to_string(count) + s[i];
                count = 1;
            }
        }
        return res;
    }
};
// @lc code=end

File: 0001-0500/0038/0038.count-and-say.py
#
# @lc app=leetcode id=38 lang=python3
#
# [38] Count and Say
#

# @lc code=start
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return '1'
        else:
            return self.count(self.countAndSay(n - 1))
    def count(self, s):
        res = ''
        i = 0
        while i < len(s):
            j = i
            while j < len(s) and s[j] == s[i]:
                j += 1
            res += str(j - i) + s[i]
            i = j
        return res
# @lc code=end

